components/camera/videosourcewrappers.go

```diff
@@ -0,0 +1,214 @@
+	}
+
+	actualSystem := syst
+	if actualSystem == nil {
+		//nolint:staticcheck
+		srcCam, ok := source.(Camera)
+		if ok {
+			props, err := srcCam.Properties(ctx)
+			if err != nil {
+				return nil, NewPropertiesError("source camera")
+			}
+			var cameraModel transform.PinholeCameraModel
+			cameraModel.PinholeCameraIntrinsics = props.IntrinsicParams
+
+			if props.DistortionParams != nil {
+				cameraModel.Distortion = props.DistortionParams
+			}
+
+			actualSystem = &cameraModel
+		}
+	}
+	return &videoSource{
+		system:       actualSystem,
+		videoSource:  source,
+		videoStream:  gostream.NewEmbeddedVideoStream(source),
+		actualSource: source,
+		imageType:    imageType,
+	}, nil
+}
+
+// videoSource implements a Camera with a gostream.VideoSource.
+type videoSource struct {
+	rtpPassthroughSource rtppassthrough.Source
+	videoSource          gostream.VideoSource
+	videoStream          gostream.VideoStream
+	actualSource         interface{}
+	system               *transform.PinholeCameraModel
+	imageType            ImageType
+}
+
+func (vs *videoSource) Stream(ctx context.Context, errHandlers ...gostream.ErrorHandler) (gostream.VideoStream, error) {
+	return vs.videoSource.Stream(ctx, errHandlers...)
+}
+
+// Images is for getting simultaneous images from different sensors
+// If the underlying source did not specify an Images function, a default is applied.
+// The default returns a list of 1 image from ReadImage, and the current time.
+func (vs *videoSource) Images(ctx context.Context) ([]NamedImage, resource.ResponseMetadata, error) {
+	ctx, span := trace.StartSpan(ctx, "camera::videoSource::Images")
+	defer span.End()
+	if c, ok := vs.actualSource.(ImagesSource); ok {
+		return c.Images(ctx)
+	}
+	img, release, err := ReadImage(ctx, vs.videoSource)
+	if err != nil {
+		return nil, resource.ResponseMetadata{}, errors.Wrap(err, "videoSource: call to get Images failed")
+	}
+	defer func() {
+		if release != nil {
+			release()
+		}
+	}()
+	ts := time.Now()
+	return []NamedImage{{img, ""}}, resource.ResponseMetadata{CapturedAt: ts}, nil
+}
+
+// NextPointCloud returns the next PointCloud from the camera, or will error if not supported.
+func (vs *videoSource) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error) {
+	ctx, span := trace.StartSpan(ctx, "camera::videoSource::NextPointCloud")
+	defer span.End()
+	if c, ok := vs.actualSource.(PointCloudSource); ok {
+		return c.NextPointCloud(ctx)
+	}
+	if vs.system == nil || vs.system.PinholeCameraIntrinsics == nil {
+		return nil, transform.NewNoIntrinsicsError("cannot do a projection to a point cloud")
+	}
+	img, release, err := vs.videoStream.Next(ctx)
+	defer release()
+	if err != nil {
+		return nil, err
+	}
+	dm, err := rimage.ConvertImageToDepthMap(ctx, img)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot project to a point cloud")
+	}
+	return depthadapter.ToPointCloud(dm, vs.system.PinholeCameraIntrinsics), nil
```

Comment:
I actually think this function is doing the heavy lifting.

Reply:
The depthadapter package is an important function for generating very fast point clouds from a depth map.

Reply:
Yes, I believe this is what is really producing point clouds. Projector wasn't doing much.